src/CommunityPatch/Patches/Perks/Intelligence/Engineering/BallisticsPatch.cs

```diff
@@ -0,0 +1,159 @@
+    }
+
+    public override void Apply(Game game) {
+      var textObjStrings = TextObject.ConvertToStringList(
+        new List<TextObject> {
+          _perk.Name,
+          _perk.Description
+        }
+      );
+      // most of the properties of skills have private setters, yet Initialize is public
+      _perk.Initialize(
+        textObjStrings[0],
+        textObjStrings[1],
+        _perk.Skill,
+        (int) _perk.RequiredSkillValue,
+        _perk.AlternativePerk,
+        _perk.PrimaryRole, _perk.PrimaryBonus,
+        _perk.SecondaryRole, _perk.SecondaryBonus,
+        _perk.IncrementType
+      );
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TooltipTargetMethodInfo, postfix: new HarmonyMethod(TooltipPatchMethodInfo));
+      CommunityPatchSubModule.Harmony.Patch(BombardTargetMethodInfo, new HarmonyMethod(BombardPatchMethodInfo));
+      Applied = true;
+    }
+
+    // ReSharper disable once InconsistentNaming
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void TooltipPostfix(ref List<TooltipProperty> __result, SiegeEngineType engine) {
+      var perk = ActivePatch._perk;
+
+      if (!IsCatapult(engine)) return;
+      if (!DefendersFromHeroSiegeHaveBallisticPerk()) return;
+
+      var emptyTooltip = __result?.LastOrDefault();
+      if (emptyTooltip == null) return;
+      emptyTooltip.DefinitionLabel = perk.Name.ToString();
+      emptyTooltip.ValueLabel = $"{(perk.PrimaryBonus * 100):F1}%";
+
+      var rangedAttackProperty = FindRangedAttackTooltipProperty(__result);
+      if (rangedAttackProperty == null) return;
+      var rangedDamage = (int)Convert.ToDouble(rangedAttackProperty.ValueLabel);
+      var totalDamage = rangedDamage + rangedDamage * perk.PrimaryBonus;
+      rangedAttackProperty.ValueLabel = $"{totalDamage:F1}";
+    }
+    
+    // ReSharper disable once InconsistentNaming
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void BombardPrefix(ISiegeEventSide siegeEventSide, SiegeEngineType attackerEngineType, SiegeEvent.SiegeEngineConstructionProgress damagedEngine) {
+      var perk = ActivePatch._perk;
+
+      if (!IsCatapult(attackerEngineType)) return;
+      if (!DefendersHaveBallisticPerk(siegeEventSide.SiegeParties)) return;
+      
+      var bonusDamage = attackerEngineType.Damage * perk.PrimaryBonus;
+      damagedEngine.SetHitpoints(damagedEngine.Hitpoints - bonusDamage);
+    }
+
+    private static bool IsCatapult(SiegeEngineType engine)
+      => engine == DefaultSiegeEngineTypes.Catapult || engine == DefaultSiegeEngineTypes.FireCatapult;
+    
+    private static bool DefendersFromHeroSiegeHaveBallisticPerk() {
+      var perk = ActivePatch._perk;
+      
+      if (Hero.MainHero == null) return false;
+      
+      if (Hero.MainHero.PartyBelongedTo.SiegeEvent == null || Hero.MainHero.PartyBelongedTo.Party.Side == BattleSideEnum.Defender) 
```

Comment:
Should we not be checking for the party's effective Engineer instead of just the leader hero?

Reply:
According to the initialization the perk has the personal role. That might be wrong. Should probably be party leader or engineer.

Reply:
Responding you tynakuh, that will depend on the perk role. For now it is 'Personal'. Should I change it?

Reply:
Ah right, hmm. That's a strange perk role, not sure what the best approach is here, Engineer makes the most logical sense to me, but party leader is valid as well. Personal makes no real sense to me (as it doesn't apply to the 'person' specifically...unless we're talking in combat usage of the catapult?)

Reply:
I'm happy for this to go in as Party Leader wihtout logic change, so long as we change the perk description/property alongside to make it clear to players.

Reply:
I have changed the role to Party Leader as suggested. Also I fixed the checking for party leader in the defensor team for displaying the Tooltip. In this case, when the player was in the defensor team, the code was only checking if the hero had the perk active. Now it checks every defender party before displaying the tooltip.